src/graph/gk_ops.cc

```diff
@@ -0,0 +1,108 @@
+/*!
+ *  Copyright (c) 2020 by Contributors
+ * \file graph/gk_ops.cc
+ * \brief Graph operation implemented in GKlib
+ */
+
+#if !defined(_WIN32)
+#include <GKlib.h>
+#endif  // !defined(_WIN32)
+
+#include <dgl/graph_op.h>
+
+namespace dgl {
+
+#if !defined(_WIN32)
+
+namespace {
+
+gk_csr_t *Convert2GKCsr(CSRPtr csr, bool is_row) {
+  // TODO(zhengda) The conversion will be zero-copy in the future.
+  const aten::CSRMatrix mat = csr->ToCSRMatrix();
+  const dgl_id_t *indptr = static_cast<dgl_id_t*>(mat.indptr->data);
+  const dgl_id_t *indices = static_cast<dgl_id_t*>(mat.indices->data);
+
+  gk_csr_t *gk_csr = gk_csr_Create();
+  gk_csr->nrows = mat.num_rows;
+  gk_csr->ncols = mat.num_cols;
+  uint64_t nnz = csr->NumEdges();
+  auto gk_indptr = gk_csr->rowptr;
+  auto gk_indices = gk_csr->rowind;
```

Comment:
Initialize them to `nullptr` as you will override in both branches below.

Reply:
i don't want to define the type explicitly.